### PR TITLE
Feature Game-Lobby/Multiplayer 

### PR DESCRIPTION
**Pull request by marte1337**
This is a rather large request, but it wasn´t really possible to break it down into smaller peices. There is a lot of refactoring to be done  which is next on my list.

### CODE REVIEW

File: pages/lobby.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+            {chats.map((chat, id) => (
+              <StyledMessage key={id}>
+                <small>{chat.username}:</small> {chat.message}
+              </StyledMessage>
+            ))}
+          </div>
+
+          <div>
+            <form onSubmit={handleSubmit}>
+              <StyledInput
```
Comment (NikRoe) [inline pages/lobby.js:130]: this input needs a label

File: pages/lobby.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+              />
+              <StyledButton type="submit">Send</StyledButton>
+            </form>
+          </div>
+        </StyledChat>
+      </div>
+      <div>
+        PLAYER ONE: <br />
+        <Link href={`/multiplayer/${username}`}>
+          <StyledButton type="text">Create Game</StyledButton>
```
Comment (NikRoe) [inline pages/lobby.js:144]: this should be a paragraph

File: pages/lobby.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        </Link>
+        <br />
+        PLAYER TWO: <br />
+        <form onSubmit={handleJoinPlayer}>
+          <StyledInput type="text" placeholder="Player you want to join..." />
```
Comment (NikRoe) [inline pages/lobby.js:149]: this would be even better with a label

File: pages/lobby.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+  border-radius: 5px;
+  padding: 10px 1rem;
+  margin: 5px 10px;
+  h2 {
+    margin: 0;
+    font-weight: 900;
+    font-size: x-large;
+    letter-spacing: 5px;
+  }
+  h4 {
```
Comment (NikRoe) [inline pages/lobby.js:181]: since you'll change it above as well ;) 

```suggestion
  h3 {
```

File: pages/lobby.js [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    font-weight: 600;
+    margin: 0.6rem;
+  }
+`;
+
+const StyledMessage = styled.div`
```
Comment (NikRoe) [inline pages/lobby.js:187]:
```suggestion
const StyledMessage = styled.p`
```

File: pages/multiplayer/[slug]/index.js [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+  const [fenHistory, setFenHistory] = useState([]);
+  const [boardOrientation, setBoardOrientation] = useState("white");
+  const [showReplayBoard, setShowReplayBoard] = useState(false);
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const [showChat, setShowChat] = useState(true);
+  const [messageToSend, setMessageToSend] = useState("");
+  const [chatStorage, setChatStorage] = useState([]);
+  const [onlineUsersCount, setOnlineUsersCount] = useState(0);
+  const [onlineUsers, setOnlineUsers] = useState([]);
```
Comment (NikRoe) [inline pages/multiplayer/[slug]/index.js:32]: is looks like onlineUsersCount can be derived from onlineUsers and doesn't need to be its own state:

```suggestion
  const [onlineUsers, setOnlineUsers] = useState([]);
  
  const onlineUsersCount = onlineUsers.length
```

File: pages/multiplayer/[slug]/index.js [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+        )}
+        <PlayerNameDisplay playerName={username} oppenentName={oppenentName} />
+
+        {!showReplayBoard && moveStatus.gameOver && (
+          <>
+            <StyledButton onClick={handleShowReplayBoard}>
+              Game Replay
+            </StyledButton>
+            <Link href="/prelobby">
+              <StyledButton type="text">Main Menu</StyledButton>
```
Comment (NikRoe) [inline pages/multiplayer/[slug]/index.js:235]: this should be a paragraph

File: pages/multiplayer/[slug]/index.js [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+              <div>
+                {chatStorage.map((data, index) => (
+                  <StyledMessage key={index}>
+                    <small>{data.username}:</small> {data.message}
+                  </StyledMessage>
+                ))}
+              </div>
+              <div>
+                <form onSubmit={handleSubmit}>
+                  <StyledInput
```
Comment (NikRoe) [inline pages/multiplayer/[slug]/index.js:275]: a label would look really good here

File: pages/prelobby.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+      </h2>
+      <h2>
+        TOTALLY <i>ZOMBIFIED</i> CHESS
+      </h2>
+      <StyledPlayerName>{username}</StyledPlayerName>
+      <StyledSection>
+        <h3>CHOOSE YOUR BOARD:</h3>
+        <div>
+          <Link href="/tutorials">
+            <StyledButton type="text">Tutorials</StyledButton>
```
Comment (NikRoe) [inline pages/prelobby.js:21]: paragraph ;)

File: pages/prelobby.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+          </Link>
+        </div>
+        <div>
+          <Link href="/singleplayer">
+            <StyledButton type="text">Player VS RandomMoveMachine</StyledButton>
```
Comment (NikRoe) [inline pages/prelobby.js:26]: paragraph ;)

File: pages/prelobby.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+          </Link>
+        </div>
+        <div>
+          <Link href="/lobby">
+            <StyledButton type="text">Enter Game Lobby</StyledButton>
```
Comment (NikRoe) [inline pages/prelobby.js:31]: paragraph ;)

**Review by NikRoe — CHANGES_REQUESTED** [review]
HI Marte,

I am really impressed that you implemented a proper 1v1. Well done!

I left you a few notes to take a closer look into, mainly regarding:
- derived state
- buttons as children of anchor tags (you should not do that for accessibility and semantics)
- inputs really like having labels (also for accessibility)

Keep it up

File: pages/multiplayer/[slug]/index.js (hunk 7 above)
Comment (marte1337) [reply pages/multiplayer/[slug]/index.js:235]: Hi Niklas, I don´t really think I understand what you mean, should i wrap the whole thing in a p tag? And what lines exactly?

Comment (NikRoe) [reply pages/multiplayer/[slug]/index.js:235]: ah, my bad, not clear enough! I seem to have highlighted too many lines, it is just for line 235

It should not be a button but a paragraph instead

Comment (marte1337) [reply pages/multiplayer/[slug]/index.js:235]: Hi Niklas, is there any other solution for better accessibility and semantics you can think of, that involves leaving the button there? It seems to be almost impossible to style the p-tag exactly the same way as the button next to it. Would using a role attribute for the button be suitable enough?

Comment (NikRoe) [reply pages/multiplayer/[slug]/index.js:235]: No, unfortunately, there isn't. You could leave the paragraph out and just go with
```js
<Link href="prelobby">Main Menu</Link>
```
Then you would only need to style the Link Component

Maybe you have the option to style both the paragraph/link and the button and find the "kleinste gemeinsamer Nenner" where both look identical.

**Pull request merged**